app/scaffold-controllers/navigationController.js

```diff
@@ -88,7 +90,14 @@ function(
     };
 
     NavigationController.prototype.back = function() {
-        this.navigationView.back();
+        var self = this;
+        self.navigationView.canGoBack().then(function(canGoBack) {
+            if (canGoBack) {
+                self.navigationView.back();
+            } else {
+                Application.closeApp();
```

Comment:
After discussing with @tony-dinh in person: There is no cart modal currently implemented in scaffold for Android. I have added a comment to the Scaffold cart modal jira ticket to ensure that the modal can be closed using the back button in Android.